apps/frontend/src/pages/CreateCharity.tsx

```diff
@@ -0,0 +1,651 @@
+  const updateOrganizationPayload = (
+    index: number,
+    key: keyof OrganizationObject,
+    value: string | number | boolean,
+  ) => {
+    setNewCharityObject((prevState) => {
+      const updatedOrganizations = [...prevState.organizations];
+      try {
+        updatedOrganizations[index] = {
+          ...updatedOrganizations[index],
+          [key]: value,
+        };
+      } catch (error) {
+        const newOrganization: OrganizationObject = {
+          name: "",
+          logoUrl: "",
+          donated: 0,
+          receiving: false,
+        };
+        updatedOrganizations.push(newOrganization);
+      }
+
+      return {
+        ...prevState,
+        organizations: updatedOrganizations,
+      };
+    });
+  };
+
+  const updateCharityName = (event: ChangeEvent<HTMLInputElement>) => {
+    updateNewCharityPayload("name", event.target.value);
+  };
+
+  const updateDescription = (event: ChangeEvent<HTMLInputElement>) => {
+    updateNewCharityPayload("description", event.target.value);
+  };
+
+  const updateStartDate = (event: ChangeEvent<HTMLInputElement>) => {
+    const dateValue = event.target.value;
+    setStartDate(dateValue);
+    if (dateValue) {
+      const dateStart = new Date(dateValue);
+      const dateEnd = new Date(dateStart);
+      dateEnd.setDate(dateStart.getDate() + 31);
+      updateNewCharityPayload("startDate", dateStart.toISOString());
+      updateNewCharityPayload("endDate", dateEnd.toISOString());
+    }
+  };
+
+  const updateCharityImage = async (event: ChangeEvent<HTMLInputElement>) => {
+    if (event.target.files) {
+      setLogoImage(event.target.files[0]);
+      const imageString = await imageBlobToBase64(event.target.files[0]);
+      setLogoImageString(imageString);
+    }
+  };
+
+  const updateOrganizationName = (
+    event: ChangeEvent<HTMLInputElement>,
+    key: string,
+  ) => {
+    const index = indexFromKey(key);
+    if (index !== -1) {
+      updateOrganizationPayload(index, "name", event.target.value);
+    }
+  };
+
+  const updateOrganizationDonated = (
+    event: ChangeEvent<HTMLInputElement>,
+    key: string,
+  ) => {
+    const index = indexFromKey(key);
+    if (index !== -1) {
+      updateOrganizationPayload(
+        index,
+        "donated",
+        parseInt(event.target.value, 10),
+      );
+    }
+  };
+
+  const updateOrganizationReceiving = (
+    event: ChangeEvent<HTMLInputElement>,
+    key: string,
+  ) => {
+    const index = indexFromKey(key);
+    if (index !== -1) {
+      updateOrganizationPayload(index, "receiving", event.target.checked);
+    }
+  };
+
+  // Finds the index given a key(org name)
+  const indexFromKey = (key: string): number => {
+    let i = 0;
+    for (; i < newCharityObject.organizations.length; i++) {
+      if (newCharityObject.organizations[i].name == key) {
+        return i;
+      }
+    }
+    return -1;
+  };
+
+  // Removes an organization, and re-indexes the images
```

Comment:
All of this logic could be removed if images were always attached to their organization

Reply:
Yeah, it didn't work earlier(attaching the image urls to the orgs was far more confusing that I thought), it's possible I could have done something else, but I'm pretty sure this should work...